src/main/java/ca/wolfram/beta/symath/MathUtils.java

```diff
@@ -19,4 +24,31 @@ public static void print(String s, Object... o) {
         System.out.println(String.format(Locale.CANADA, s, o));
     }
 
+    public static Set<MathNode> getVariableTerms(MathNode node) {
+        Set<MathNode> set;
+        switch (node.getType()) {
+        case CONSTANT:
+        case MATH_CONSTANT:
+            set = new HashSet<MathNode>();
+            break;
+        case VARIABLE:
+            set = new HashSet<MathNode>();
+            set.add(node);
+            break;
+        default:
+            set = node.getChildren().stream().filter(children -> !children.isConstant()).collect(Collectors.toSet());
+            break;
+        }
+        return set;
+    }
+
+    public static boolean isLikeTerm(MathNode n1, MathNode n2) {
```

Comment:
Realistically I think what we should do is make sure that the same term, given node by node in any order, will always result in the same String. That way we can easily compare it, or at least portions of the expression